cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -645,7 +648,42 @@ private TwillController startTwillApplication(TwillRunnerService twillRunner) {
                                                                                 getSystemServiceInstances()));
 
         if (cConf.getBoolean(Constants.COLLECT_CONTAINER_LOGS)) {
-          preparer.addLogHandler(new PrinterLogHandler(new PrintWriter(System.out)));
+          preparer.addLogHandler(new LogHandler() {
+            @Override
+            public void onLog(LogEntry entry) {
+              if (entry.getThrowable() == null) {
+                LOG.info(entry.getMessage());
```

Comment:
It might be simpler to create an `ch.qos.logback.classic.spi.ILoggingEvent` from `LogEntry`, and then call `ch.qos.logback.classic.Logger.callAppenders(ILoggingEvent event)`. Then you wouldn't have to do all the stacktrace manipulation done below I think.